app/queries/get_application_choices_with_full_courses.rb

```diff
@@ -0,0 +1,11 @@
+class GetApplicationChoicesWithFullCourses
+  attr_reader :application_form
+
+  def initialize(application_form:)
+    @application_form = application_form
+  end
+
+  def call
+    application_form.application_choices.each.with_object([]) { |choice, array| array << choice.course if choice.course_option.no_vacancies? }
```

Comment:
this returns an array of `Course` rather than `ApplicationChoice`. is that what's wanted?